pkg/controllers/nodeclass/status/subnet.go

```diff
@@ -44,55 +47,77 @@ func NewSubnetReconciler(subnetClient instance.SubnetsAPI) *SubnetReconciler {
 const (
 	SubnetUnreadyReasonNotFound = "SubnetNotFound"
 
-	//TODO: Add additional case that checks that subscription, rg, and vnet are the same for incoming
-	// subnet-ids
 	SubnetUnreadyReasonIDInvalid = "SubnetIDInvalid"
 
 	// TODO(bsoghigian): Support SubnetFull readiness reason
 	SubnetUnreadyReasonFull = "SubnetFull"
 
+	// TODO(bsoghigian): check if the cidr of a subnet-id is overlapping with any of the static agentpools,
+	// AKSNodeClass subnets, or any defaulting reserved networking addresses for AKS (--dns-service-ip)
+	SubnetUnreadyReasonCIDROverlapping = "SubnetCIDROverlapping"
+
+	// TODO(bsoghigian): check cluster identity has rbac for subnet/read subnet/join for a given vnetSubnetID
+	SubnetUnreadyReasonRBACInvalid = "SubnetRBACInvalid"
+)
+
+const (
+	// TODO: Use this in SubnetFull logic
 	// Azure reserves the first four addresses and the last address, for a total of 5 ips for each subnet.
 	// For example 192.168.1.0/24 has the following reserved addresses
 	// 192.168.1.0 Network Address
 	// 192.168.1.1 Reserved by azure for the default gateway
 	// 192.168.1.2, 192.168.1.3: Reserved by Azure to map the Azure DNS IP addresses to the virtual network space.
 	// 192.168.1.255: Network broadcast address
 	AzureReservedIPs = 5
-
-	// TODO(bsoghigian): check if the cidr of a subnet-id is overlapping with any of the static agentpools,
-	// AKSNodeClass subnets, or any defaulting reserved networking addresses for AKS (--dns-service-ip)
-	SubnetUnreadyReasonCIDROverlapping = "SubnetCIDROverlapping"
-
-	// TODO(bsoghigian): check cluster identity has rbac for subnet/read subnet/join for a given vnetSubnetID
-	SubnetUnreadyReasonRBACInvalid = "SubnetRBACInvalid"
 )
 
 func (r *SubnetReconciler) Reconcile(ctx context.Context, nodeClass *v1beta1.AKSNodeClass) (reconcile.Result, error) {
-	return r.validateVNETSubnetID(ctx, nodeClass)
 	// TODO: Handle podSubnetID readiness here as well
-	// Access to state from the cluster only needs to be retrieved once for cidr validation
+	result, err := r.validateVNETSubnetID(ctx, nodeClass)
+	return result, err
 }
 
 func (r *SubnetReconciler) validateVNETSubnetID(ctx context.Context, nodeClass *v1beta1.AKSNodeClass) (reconcile.Result, error) {
+	logger := log.FromContext(ctx)
+	clusterSubnetID := options.FromContext(ctx).SubnetID
 	subnetID := lo.Ternary(nodeClass.Spec.VNETSubnetID != nil, lo.FromPtr(nodeClass.Spec.VNETSubnetID), options.FromContext(ctx).SubnetID)
```

Comment:
Potentially add logger values for subnetid, and maybe clustersubnetId?

Example:
https://github.com/Azure/karpenter-provider-azure/blob/c25dfd40f244954f1af051066e3047cb4017b6b0/pkg/controllers/nodeclass/status/kubernetesversion.go#L74